metaflow/decorators.py

```diff
@@ -469,11 +473,26 @@ def _attach_decorators_to_step(step, decospecs):
 def _init_flow_decorators(
     flow, graph, environment, flow_datastore, metadata, logger, echo, deco_options
 ):
+    # Certain decorators can be specified multiple times and exist as lists in the _flow_decorators dictionary
     for deco in flow._flow_decorators.values():
-        opts = {option: deco_options[option] for option in deco.options}
-        deco.flow_init(
-            flow, graph, environment, flow_datastore, metadata, logger, echo, opts
-        )
+        if type(deco) == list:
+            for rd in deco:
+                opts = {option: deco_options[option] for option in rd.options}
+                rd.flow_init(
+                    flow,
+                    graph,
+                    environment,
+                    flow_datastore,
+                    metadata,
+                    logger,
+                    echo,
+                    opts,
+                )
+        else:
```

Comment:
as a small nit, maybe we just always make it a list. It would simplify this and the other change. A few more changes would be required when `_flow_decorators` is accessed but IMHO, it is nicer if there is a single type and we don't have to assume the number of decorators of a given type.